libminifi/src/core/yaml/YamlConfiguration.cpp

```diff
@@ -698,6 +703,34 @@ void YamlConfiguration::checkRequiredField(YAML::Node *yamlNode, const std::stri
   }
 }
 
+YAML::Node YamlConfiguration::getOptionalField(YAML::Node *yamlNode,
+                                               const std::string &fieldName,
+                                               const YAML::Node &defaultValue,
+                                               const std::string &yamlSection,
+                                               const std::string &providedInfoMessage) {
+  std::string infoMessage = providedInfoMessage;
+  auto result = yamlNode->as<YAML::Node>()[fieldName];
+  if (!result) {
+    if (infoMessage.empty()) {
+      // Build a helpful info message for the user to inform them that a default is being used
+      infoMessage =
+          yamlNode->as<YAML::Node>()["name"] ?
+          "Using default value for optional field '" + fieldName + "' in component named '"
+              + yamlNode->as<YAML::Node>()["name"].as<std::string>() + "'" :
+          "Using default value for optional field '" + fieldName + "' ";
+      if (!yamlSection.empty()) {
+        infoMessage += " [in '" + yamlSection + "' section of configuration file]: ";
+      }
+
+      infoMessage += defaultValue.as<std::string>();
+    }
+    logger_->log_info(infoMessage.c_str());
+    result = defaultValue;
```

Comment:
no return of the result. please run make linter

Reply:
My bad. Will fix & re-push.

I didn't notice any issues in make linter, though.